src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -534,22 +535,17 @@ public boolean equals(Object obj) {
 		if (openShiftVersion == null) {
 			if (other.openShiftVersion != null)
 				return false;
-		} else if (!openShiftVersion.equals(other.openShiftVersion))
+		} else if (!openShiftVersion.equals(other.openShiftVersion)) {
 			return false;
-		//check user token to distinguish 2 clients from the same server
+		}
 		if (strategy == null) {
-			if (other.strategy != null)
-				return false;
+			return other.strategy == null;
 		} else {
-			String token = strategy.getToken();
-			String otherToken = other.strategy == null?null:other.strategy.getToken();
-			if (token == null) {
-				if (otherToken != null)
-					return false;
-			} else if (!token.equals(otherToken))
+			if (other.strategy == null) {
 				return false;
+			}
+			return ObjectUtils.equals(strategy, other.strategy);
```

Comment:
DefaultClient#equals wont rely on the token, only on strategy equality.
Strategies only rely on the username for equality (see comment further down).